app/models/page/DashboardPage.scala

```diff
@@ -16,111 +16,251 @@
 
 package models.page
 
-import models.ApplicationData.ApplicationStatus
-import models.ApplicationData.ApplicationStatus.ApplicationStatus
-import models.page.DashboardPage.ProgressStepVisibility
-import models.{ ApplicationData, CachedData, Progress }
+import connectors.{ AllocationExchangeObjects, ExchangeObjects }
+import models.page.DashboardPage.Flags.{ ProgressInactive, _ }
+import models.{ CachedData, Progress }
+import org.joda.time.LocalDate
 import play.api.i18n.Lang
 import play.api.mvc.RequestHeader
-import security.RoleUtils
-import security.Roles.DisplayOnlineTestSectionRole
+import security.RoleUtils._
+import security.Roles._
 
-// format: OFF
 case class DashboardPage(firstStepVisibility: ProgressStepVisibility,
                          secondStepVisibility: ProgressStepVisibility,
                          thirdStepVisibility: ProgressStepVisibility,
-                         fourthStepVisibility: ProgressStepVisibility)
-
-// format: ON
+                         fourthStepVisibility: ProgressStepVisibility,
+                         isApplicationSubmittedAndNotWithdrawn: Boolean,
+                         isApplicationInProgressAndNotWithdrawn: Boolean,
+                         isApplicationWithdrawn: Boolean,
+                         isApplicationCreatedOrInProgress: Boolean,
+                         isUserWithNoApplication: Boolean,
+                         isFirstStepVisible: String,
```

Comment:
Not sure why these are strings? A boolean seems more appropriate, given the name. Also, is there even a need to have these given the parameters on line 29-32 (e.g. firstStepVisibility: ProgressStepVisibility)